src/org/labkey/test/components/domain/DomainFieldRow.java

```diff
@@ -1185,6 +1209,11 @@ protected class ElementCache extends WebDriverComponent.ElementCache
                 .refindWhenNeeded(this));
         public final Input charScaleInput = new Input(Locator.tagWithAttributeContaining("input", "id", "domainpropertiesrow-scale")
                 .refindWhenNeeded(this), getDriver());
+
+        // barcode option
+        public final Locator scannableCheckboxLoc = Locator.input("domainpropertiesrow-scannable");
+        public final Checkbox scannableCheckbox = new Checkbox(scannableCheckboxLoc.findWhenNeeded(this).withTimeout(WAIT_FOR_JAVASCRIPT));
```

Comment:
Shouldn't this be `refindWhenNeeded`? It's possible to change the field type and the checkbox will go away, and then change it back to a supported type and it will reappear.